### PR TITLE
Add series on DP

### PR DESCRIPTION
**Pull request by ganesh-sadanala**
Please read the notes on dp, comment the bugs in it, contribute your tips,tricks,ideas,code to it. I will be adding a lot to it.

### CODE REVIEW

File: Java/Dynamic-Programming/DP.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                return Min[sum];				
+			}				
+		 }
+	   }
+	   public static void main(String []args)
+	   {
+		   Coins c = new Coins(3);
+		   System.out.println("Minimum number of coins for the given sum 4: "+c.minCoins(4));
+	   }
+	}
```
Comment (samagragupta) [inline Java/Dynamic-Programming/DP.java]: Please have an EOF

Comment (ganesh-sadanala) [reply Java/Dynamic-Programming/DP.java]: Thanks! it is a good practice to have EOF. I have added it

File: Java/Dynamic-Programming/DP.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  Same value is calculated number of times incurring cpu cycles and time cost as well as stack resource.
+Two ways we can eliminate this:
+1.Memoization/TopDown - Look in the data structure(a look up table in the memory ),
+if present
+  return that value
+else
+  find and place in the look up table
+
+2nd method will be dealt after some hands on coding
+*/
```
Comment (Gurubalan-GIT) [inline Java/Dynamic-Programming/DP.java:25]: Damn nice job on the comments and sticking with Camel Case! 👍

**Review by Gurubalan-GIT — COMMENTED** [review]
Amazing job, If you are solving more problems (Competetive aimed) do actually push it into the folder named `Competetive- coding` and have the source links in the comment lines (Like hackerrank etc. ) 👍

File: Java/Dynamic-Programming/DP.java (hunk 2 above)
Comment (ganesh-sadanala) [reply Java/Dynamic-Programming/DP.java:25]: Thanks buddy! there are a lot to come.

Comment (ganesh-sadanala) [reply Java/Dynamic-Programming/DP.java:25]: Sure! I will comment the links to amazing competitive problems.

**Pull request merged**